arch/x86/src/qemu/qemu_saveusercontext.S

```diff
@@ -132,16 +132,16 @@ up_saveusercontext:
 	 *							EIP
 	 *							CS					...
 	 *							EFLAGS				EIP
-	 *						->	ESP					CS
+	 *							ESP					CS
 	 * ESP->Return address		SS					EFLAGS
-	 *		Argument			Argument			Argument
+	 *		Argument		->	Argument			Argument
```

Comment:
Saved %esp must point here. Because up_fullcontextrestore creates the IRET stack frame after restoring %esp.